framework/bundles/org.jboss.tools.rsp.secure/src/org/jboss/tools/rsp/secure/model/RSPSecureStorage.java

```diff
@@ -0,0 +1,169 @@
+package org.jboss.tools.rsp.secure.model;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.jboss.tools.rsp.secure.crypto.CryptoException;
+import org.jboss.tools.rsp.secure.crypto.CryptoUtils;
+import org.jboss.tools.rsp.secure.crypto.NotInitializedCryptoException;
+
+public class RSPSecureStorage implements ISecureStorage {
+	private static final String COMMENT = "rsp secure storage";
+	private static final String HASH_COMMENT = "#" + COMMENT;
```

Comment:
I'd include in this comment the current algorithm for latter identification

Reply:
@jeffmaury discussing this makes me discover that this isn't cutting: the whole file is encrypted. The comment is there to have a fixed entry that allows the code to identify if the decrypt worked. In other words you need to know the algorithm before you decrypt. 
That's of course not the case if an unencrypted String is present in the beginning of the file that identifies the algorithm.